extensions/binding/org.eclipse.smarthome.binding.mqtt.automation/src/main/java/org/eclipse/smarthome/automation/module/mqtt/internal/MQTTModuleTypeProvider.java

```diff
@@ -0,0 +1,215 @@
+                "Publish MQTT message", "Publishes a message to an MQTT topic", //
+                null, Visibility.VISIBLE, getInputActionDesc(locale), getOutputTriggerDesc(locale));
+        this.actionType = actionType;
+        return actionType;
+    }
+
+    private ModuleType getTriggerType(Locale locale) {
+        return new TriggerType(PublishedMessageTriggerHandler.MODULE_TYPE_ID, getConfigTriggerDesc(locale),
+                "MQTT message published", "Triggers whenever the given MQTT topic has a new value", //
+                null, Visibility.VISIBLE, getOutputTriggerDesc(locale));
+    }
+
+    private @Nullable List<Output> getOutputTriggerDesc(Locale locale) {
+        Output topicName = new Output(MQTTModuleConstants.INOUT_TOPIC_NAME, MQTTModuleConstants.TYPE_TOPIC, //
+                "Topic", "A topic that was triggered from", null, null, null);
+        Output topicValue = new Output(MQTTModuleConstants.INOUT_TOPIC_VALUE, MQTTModuleConstants.TYPE_STRING, //
+                "Topic", "The received topic value if triggered from a topic", null, null, null);
+        Output brokerConnection = new Output(MQTTModuleConstants.INOUT_BROKER_CONNECTION,
+                MQTTModuleConstants.INOUT_BROKER_CONNECTION, //
+                "Broker connection", "The broker connection", null, null, null);
+        return Stream.of(topicName, topicValue, brokerConnection).collect(Collectors.toList());
+    }
+
+    private @Nullable List<Input> getInputActionDesc(Locale locale) {
+        Input topicName = new Input(MQTTModuleConstants.INOUT_TOPIC_NAME, MQTTModuleConstants.TYPE_TOPIC, //
+                "Topic", "A topic that was triggered from", null, false, null, null);
+        Input topicValue = new Input(MQTTModuleConstants.INOUT_TOPIC_VALUE, MQTTModuleConstants.TYPE_STRING, //
+                "Topic", "The received topic value if triggered from a topic", null, false, null, null);
+        Input brokerConnection = new Input(MQTTModuleConstants.INOUT_BROKER_CONNECTION,
+                MQTTModuleConstants.INOUT_BROKER_CONNECTION, //
+                "Broker connection", "The broker connection", null, false, null, null);
+        return Stream.of(topicName, topicValue, brokerConnection).collect(Collectors.toList());
+    }
+
+    private List<ConfigDescriptionParameter> getConfigActionDesc(Locale locale) {
+        ConfigDescriptionParameter paramBroker = ConfigDescriptionParameterBuilder
+                .create(MQTTModuleConstants.CFG_BROKER, Type.TEXT).withRequired(true).withLabel("Broker")
+                .withDescription("The broker to publish to").withOptions(getBrokerConnectionIDs())
+                .withLimitToOptions(true).build();
+        ConfigDescriptionParameter paramMessage = ConfigDescriptionParameterBuilder
+                .create(MQTTModuleConstants.CFG_MESSAGE, Type.TEXT).withRequired(true).withLabel("Message")
```

Comment:
The message (topicValue) is already an input of the action, so it should not be a configuration parameter as well.

Reply:
I'm still confused about configuration parameters and inputs. Where can I specify inputs in PaperUI to test my rule actions?